common/src/services/userVerification.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+
+import { UserVerificationService as UserVerificationServiceAbstraction } from '../abstractions/userVerification.service';
+
+import { ApiService } from '../abstractions/api.service';
+import { CryptoService } from '../abstractions/crypto.service';
+import { I18nService } from '../abstractions/i18n.service';
+import { LogService } from '../abstractions/log.service';
+import { PlatformUtilsService } from '../abstractions/platformUtils.service';
+
+import { VerificationType } from '../enums/verificationType';
+
+import { VerifyOtpRequest } from '../models/request/account/verifyOtpRequest';
+import { PasswordVerificationRequest } from '../models/request/passwordVerificationRequest';
+
+import { Verification } from '../types/verification';
+
+@Injectable()
+export class UserVerificationService implements UserVerificationServiceAbstraction {
+    constructor(private cryptoService: CryptoService, private i18nService: I18nService,
+        private platformUtilsService: PlatformUtilsService, private apiService: ApiService,
+        private logService: LogService) { }
+
+    async buildRequest<T extends PasswordVerificationRequest>(verification: Verification,
+        requestClass?: new () => T, alreadyHashed?: boolean) {
+        if (verification?.secret == null || verification.secret === '') {
```

Comment:
verification?.secret == null checks for '' here, doesn't it?

Reply:
I still assume the user can submit an empty string, unless reactive forms change this to a null value?

Reply:
`null != ''`. We could change it to `if (verification?.secret)`, but that feels less explicit.